server/api/db/base.py

```diff
@@ -758,3 +758,41 @@ def list_datastore_profiles(
         project: str,
     ) -> list[mlrun.common.schemas.DatastoreProfile]:
         pass
+
+    # Pagination Cache Methods
+    # They are not abstract methods because they are not required for all DBs.
+    # However, they do raise NotImplementedError for DBs that do not implement them.
+    def store_paginated_query_cache_record(
+        self,
+        session,
+        user: str,
+        function: str,
+        current_page: int,
+        kwargs: dict,
+    ):
+        raise NotImplementedError
+
+    def get_paginated_query_cache_record(
+        self,
+        session,
+        key: str,
+    ):
+        raise NotImplementedError
+
+    def list_paginated_query_cache_record(
+        self,
+        session,
+        key: str = None,
+        user: str = None,
+        function: str = None,
+        last_accessed_before: datetime = None,
+        order_by_desc: bool = True,
+    ):
+        raise NotImplementedError
+
+    def delete_paginated_query_cache_record(
+        self,
+        session,
+        key: str,
+    ):
+        raise NotImplementedError
```

Comment:
```suggestion
        raise NotImplementedError
        # EOF Pagination Cache Methods
```
just to put the barrier